Intellij logs show guice throws

After upgrading from turbo builder extension v0.3 to v0.9 IntelliJ IDEA 2025.2.1 logs include:

```
2025-09-19 12:20:03,686 [1563476]   WARN - #o.j.i.maven - com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) [Guice/ErrorInCustomProvider]: IllegalStateException
  at SessionScopeModule.configure(SessionScopeModule.java:55)
      \_ installed by: WireModule -> SessionScopeModule
  at TurboBuilderConfig.<init>(TurboBuilderConfig.java:14)
      \_ for 1st parameter session
  while locating TurboBuilderConfig annotated with interface TypeArguments$Implicit
  at LocatorWiring
  at TurboMavenLifecycleParticipant.<init>(TurboMavenLifecycleParticipant.java:37)
      \_ for 1st parameter config
  at ClassRealm[coreExtension>com.github.seregamorph:maven-turbo-builder:0.9, parent: ClassRealm[plexus.core, parent: null]]
      \_ installed by: WireModule -> PlexusBindingModule
  while locating TurboMavenLifecycleParticipant
  while locating Object annotated with *

Learn more:
  https://github.com/google/guice/wiki/ERROR_IN_CUSTOM_PROVIDER

1 error

======================
Full classname legend:
======================
LocatorWiring:                  "org.eclipse.sisu.wire.LocatorWiring"
PlexusBindingModule:            "org.eclipse.sisu.plexus.PlexusBindingModule"
SessionScopeModule:             "org.apache.maven.session.scope.internal.SessionScopeModule"
TurboBuilderConfig:             "com.github.seregamorph.maven.turbo.TurboBuilderConfig"
TurboMavenLifecycleParticipant: "com.github.seregamorph.maven.turbo.TurboMavenLifecycleParticipant"
TypeArguments$Implicit:         "org.eclipse.sisu.inject.TypeArguments$Implicit"
WireModule:                     "org.eclipse.sisu.wire.WireModule"
========================
End of classname legend:
========================
	in InternalProvisionException.toProvisionException(InternalProvisionException.java:251)
Caused by: java.lang.IllegalStateException
	in SessionScope.lambda$static$0(SessionScope.java:38)
	... 25 more


...
```

[turbo.log](https://github.com/user-attachments/files/22425539/turbo.log)

--- a/src/main/java/com/github/seregamorph/maven/turbo/TurboBuilder.java
+++ b/src/main/java/com/github/seregamorph/maven/turbo/TurboBuilder.java
@@ -51,16 +51,20 @@ public class TurboBuilder implements Builder {
 
     public static final String BUILDER_TURBO = "turbo";
 
+    private final DefaultLifecycles defaultLifeCycles;
     private final LifecycleModuleBuilder lifecycleModuleBuilder;
 
     @Inject
     public TurboBuilder(
         DefaultLifecycles defaultLifeCycles,
-        LifecycleModuleBuilder lifecycleModuleBuilder,
-        TurboBuilderConfig config
+        LifecycleModuleBuilder lifecycleModuleBuilder
     ) {
+        this.defaultLifeCycles = defaultLifeCycles;
         this.lifecycleModuleBuilder = lifecycleModuleBuilder;
+    }
 
+    private void patchLifecycles(MavenSession session) {
+        TurboBuilderConfig config = TurboBuilderConfig.fromSession(session);
         String mavenVersion = Maven.class.getPackage().getImplementationVersion();
         // since Maven 4 changes of DefaultLifecycles have no effect
         if (mavenVersion != null && mavenVersion.startsWith("3.")) {
@@ -88,6 +92,7 @@ public class TurboBuilder implements Builder {
             session.getRequest().getDegreeOfConcurrency(),
             session.getProjects().size());
         logger.info("TurboBuilder will use {} threads to build {} modules", nThreads, session.getProjects().size());
+        patchLifecycles(session);
         boolean parallel = nThreads > 1;
         // Propagate the parallel flag to the root session and all of the cloned sessions in each project segment
         session.setParallel(parallel);

--- a/src/main/java/com/github/seregamorph/maven/turbo/TurboBuilderConfig.java
+++ b/src/main/java/com/github/seregamorph/maven/turbo/TurboBuilderConfig.java
@@ -1,6 +1,5 @@
 package com.github.seregamorph.maven.turbo;
 
-import javax.inject.Inject;
 import org.apache.maven.execution.MavenSession;
 
 /**
@@ -10,10 +9,9 @@ public class TurboBuilderConfig {
 
     private final boolean turboTestCompile;
 
-    @Inject
-    public TurboBuilderConfig(MavenSession session) {
+    static TurboBuilderConfig fromSession(MavenSession session) {
         String turboTestCompile = MavenPropertyUtils.getProperty(session, "turboTestCompile");
-        this.turboTestCompile = MavenPropertyUtils.isTrue(turboTestCompile);
+        return new TurboBuilderConfig(MavenPropertyUtils.isTrue(turboTestCompile));
     }
 
     TurboBuilderConfig(boolean turboTestCompile) {

--- a/src/main/java/com/github/seregamorph/maven/turbo/TurboMavenLifecycleParticipant.java
+++ b/src/main/java/com/github/seregamorph/maven/turbo/TurboMavenLifecycleParticipant.java
@@ -5,7 +5,6 @@ import static com.github.seregamorph.maven.turbo.MavenPropertyUtils.isTrue;
 
 import java.util.List;
 import java.util.stream.Collectors;
-import javax.inject.Inject;
 import javax.inject.Named;
 import org.apache.maven.AbstractMavenLifecycleParticipant;
 import org.apache.maven.MavenExecutionException;
@@ -31,13 +30,6 @@ public class TurboMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
     private static final Logger logger = LoggerFactory.getLogger(TurboMavenLifecycleParticipant.class);
 
-    private final TurboBuilderConfig config;
-
-    @Inject
-    public TurboMavenLifecycleParticipant(TurboBuilderConfig config) {
-        this.config = config;
-    }
-
     @Override
     public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
         if (isTurboBuilder(session)) {
@@ -54,6 +46,7 @@ public class TurboMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
     }
 
     private void checkTestJarArtifacts(MavenSession session) throws MavenExecutionException {
+        TurboBuilderConfig config = TurboBuilderConfig.fromSession(session);
         if (!config.isTurboTestCompile()) {
             // test-jar is not supported, because package phase is now executed before compiling tests
             for (MavenProject project : session.getProjects()) {
@@ -83,6 +76,7 @@ public class TurboMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
     }
 
     private void checkBuilderAndPhase(MavenSession session) {
+        TurboBuilderConfig config = TurboBuilderConfig.fromSession(session);
         // skip both compiling and running tests
         boolean mavenTestSkip = isTrue(getProperty(session, "maven.test.skip"));
         // skip only running tests

--- a/src/main/java/com/github/seregamorph/maven/turbo/TurboProjectExecutionListener.java
+++ b/src/main/java/com/github/seregamorph/maven/turbo/TurboProjectExecutionListener.java
@@ -4,7 +4,6 @@ import static com.github.seregamorph.maven.turbo.PhaseOrderPatcher.isPackage;
 
 import java.util.Objects;
 import java.util.stream.Collectors;
-import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
 import org.apache.maven.Maven;
@@ -17,13 +16,6 @@ import org.apache.maven.execution.ProjectExecutionListener;
 @Named
 @Singleton
 public class TurboProjectExecutionListener implements ProjectExecutionListener {
-
-    private final TurboBuilderConfig config;
-
-    @Inject
-    public TurboProjectExecutionListener(TurboBuilderConfig config) {
-        this.config = config;
-    }
 
     @Override
     public void beforeProjectExecution(ProjectExecutionEvent event) {
@@ -40,6 +32,7 @@ public class TurboProjectExecutionListener implements ProjectExecutionListener {
                 .collect(Collectors.toList());
 
             if (isReorderPhases()) {
+                TurboBuilderConfig config = TurboBuilderConfig.fromSession(event.getSession());
                 PhaseOrderPatcher.reorderPhases(config, event.getExecutionPlan(), mojoExecution ->
                     Objects.toString(mojoExecution.getLifecyclePhase(), ""));
             }

--- a/src/main/resources/META-INF/sisu/javax.inject.Named
+++ b/src/main/resources/META-INF/sisu/javax.inject.Named
@@ -1,5 +1,4 @@
 com.github.seregamorph.maven.turbo.TurboBuilder
-com.github.seregamorph.maven.turbo.TurboBuilderConfig
 com.github.seregamorph.maven.turbo.TurboMavenLifecycleParticipant
 com.github.seregamorph.maven.turbo.TurboMojoExecutionListener
 com.github.seregamorph.maven.turbo.TurboProjectExecutionListener

--- a/src/test/java/com/github/seregamorph/maven/turbo/TurboMojosExecutionStrategyMaven3Test.java
+++ b/src/test/java/com/github/seregamorph/maven/turbo/TurboMojosExecutionStrategyMaven3Test.java
@@ -194,8 +194,7 @@ class TurboMojosExecutionStrategyMaven3Test {
 
         var strategy = new DefaultMojosExecutionStrategy();
         var eventsList = new ArrayList<String>();
-        var turboBuilderConfig = new TurboBuilderConfig(session);
-        var turboProjectExecutionListener = new TurboProjectExecutionListener(turboBuilderConfig) {
+        var turboProjectExecutionListener = new TurboProjectExecutionListener() {
             @Override
             boolean isReorderPhases() {
                 return false;

